charts/prometheus/templates/server/sts.yaml

```diff
@@ -133,6 +133,13 @@ spec:
               path: {{ .Values.server.prefixURL }}/-/ready
               port: 9090
               scheme: {{ .Values.server.probeScheme }}
+              {{- if .Values.server.probeHeaders }}
+              httpHeaders:
+              {{- range .Values.server.probeHeaders}}
+              - name: {{ .name }}
+                value: {{ .value }}
+              {{- end }}
+              {{- end }}
```

Comment:
Why can't we use `with` statement and pump yaml directly here? e.g.
```
# sts.yaml
...
            {{- with .Values.server.probeHeaders }}
            httpHeaders:
{{ toYaml . | nindent 12 }}
            {{- end }}
...
---
# values.yaml
...
server:
  probeHeaders:
    - name: SecretAccessToken
      value: AaAbBbSuperSecret
...
```

Reply:
@zanhsieh updated with a new commit. please review